### PR TITLE
F2D immunization transformer

### PR DESCRIPTION
**Pull request by ilaflamme-va**
- F2D immunization controller
- updated _toDatamartReferenceWithCdwId_ to handle partial reference objects

### CODE REVIEW

File: data-query/src/test/java/gov/va/api/health/dataquery/tools/minimart/transformers/F2DImmunizationTransformer.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+      return null;
+    }
+    return Optional.of(note.get(0).text());
+  }
+
+  private Reference reaction(List<Reaction> reactions) {
+    if (reactions == null) {
+      return null;
+    }
+    if (reactions.isEmpty()) {
```
Comment (jhulbert-va) [inline data-query/src/test/java/gov/va/api/health/dataquery/tools/minimart/transformers/F2DImmunizationTransformer.java]: These could be mashed together using an or.

File: data-query/src/test/java/gov/va/api/health/dataquery/tools/minimart/transformers/F2DImmunizationTransformer.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            .seriesDoses(vaccinationProtocol.get(0).seriesDoses())
+            .build());
+  }
+
+  private VaccineCode vaccineCode(CodeableConcept vaccineCode) {
+    if (vaccineCode == null) {
+      return null;
+    }
+    return VaccineCode.builder()
+        .code(vaccineCode.coding() == null ? null : vaccineCode.coding().get(0).code())
```
Comment (jhulbert-va) [inline data-query/src/test/java/gov/va/api/health/dataquery/tools/minimart/transformers/F2DImmunizationTransformer.java]: Coding could be empty.

**Comment by jhulbert-va** [comment]
Diff between fhir results is flawless. After feedback is addressed, I'm good to merge.

File: data-query/src/test/java/gov/va/api/health/dataquery/tools/minimart/transformers/F2DImmunizationTransformer.java (hunk 2 above)
Comment (ilaflamme-va) [reply data-query/src/test/java/gov/va/api/health/dataquery/tools/minimart/transformers/F2DImmunizationTransformer.java]: Fixed.

**Pull request merged**